depmapomics/ccle.py

```diff
@@ -0,0 +1,1423 @@
+        na_values=["__UNKNOWN__", "."],
+        engine="c",
+        dtype=str,
+    )
+    unfiltered["somatic"] = unfiltered["somatic"].replace("nan", "False")
+    unfiltered["HGNC_Status"] = unfiltered["HGNC_Status"].replace("nan", "Unapproved")
+    unfiltered["judgement"] = unfiltered["judgement"].replace("nan", "REMOVE")
+    unfiltered = unfiltered.rename(
+        columns={
+            "i_ExAC_AF": "ExAC_AF",
+            "Tumor_Sample_Barcode": SAMPLEID,
+            "Tumor_Seq_Allele2": "Tumor_Allele",
+        }
+    ).drop(columns=["Tumor_Seq_Allele1"])
+    unfiltered["CGA_WES_AC"] = [
+        str(i[0]) + ":" + str(i[1])
+        for i in np.nan_to_num(
+            unfiltered[["t_alt_count", "t_ref_count"]].values.astype(float), 0
+        ).astype(int)
+    ]
+    toremove = []
+    subunfilt = unfiltered.iloc[:10000]
+    for i, val in enumerate(unfiltered.columns):
+        h.showcount(i, len(unfiltered.columns))
+        if len(set(subunfilt[val]) - set(["nan"])) == 1:
+            if len(set(unfiltered[val]) - set(["nan"])) == 1:
+                toremove.append(val)
+    unfiltered = unfiltered.drop(columns=set(toremove))
+    toint = ["Start_position", "End_position"]
+    for val in toint:
+        unfiltered[val] = unfiltered[val].astype(int)
+    unfiltered.to_csv(
+        folder + "mutation_somatic_unfiltered_withreplicates.csv.gz", index=False
+    )
+    unfiltered = unfiltered[subsetcol]
+    unfiltered.to_csv(
+        folder + "mutation_somatic_unfiltered_withreplicates_subseted.csv.gz",
+        index=False,
+    )
+    os.system("gunzip " + folder + "mutation_somatic_unfiltered_withreplicates.csv.gz")
+    del unfiltered
+    tc.update_dataset(
+        changes_description="adding unfiltered mutations",
+        dataset_permaname=taiga_dataset,
+        upload_files=[
+            {
+                "path": folder
+                + "mutation_somatic_unfiltered_withreplicates_subseted.csv",
+                "format": "TableCSV",
+                "encoding": "utf-8",
+            },
+            {
+                "path": folder + "mutation_somatic_unfiltered_withreplicates.csv",
+                "format": "TableCSV",
+                "encoding": "utf-8",
+            },
+        ],
+        add_all_existing_files=True,
+        upload_async=False,
+    )
+
+
+def mapBed(file, vcfdir):
+    """map mutations in one vcf file to regions in the guide bed file"""
+
+    guides_bed = pd.read_csv(
+        GUIDESBED, sep="\t", header=None, names=["chrom", "start", "end", "foldchange"]
+    )
+
+    bed = pd.read_csv(
+        vcfdir + file,
+        sep="\t",
+        header=None,
+        names=["chrom", "start", "end", "foldchange"],
+    )
+    bed["foldchange"] = 1
+    name = file.split("/")[-1].split(".")[0].split("_")[1]
+    if len(bed) == 0:
+        return (name, None)
+    val = chip.putInBed(guides_bed, bed, mergetype="sum")
+
+    return (name, val)
+
+
+def generateGermlineMatrix(
```

Comment:
same

Reply:
we want to separate this into what people would run to reproduce our results and what we want to run. most of it should be in mutations and should be called by the ccle.mutationPostProcessing script I think